resolwe_bio/processes/workflows/bbduk_star.py

```diff
@@ -0,0 +1,714 @@
+            data_type="annotation",
+            label="Annotation",
+            description="GTF and GFF3 annotation formats are supported.",
+        )
+        assay_type = StringField(
+            label="Assay type",
+            choices=[
+                ("non_specific", "Strand non-specific"),
+                ("forward", "Strand-specific forward"),
+                ("reverse", "Strand-specific reverse"),
+                ("auto", "Detect automatically"),
+            ],
+            description="In strand non-specific assay a read is considered overlapping with a "
+            "feature regardless of whether it is mapped to the same or the opposite "
+            "strand as the feature. In strand-specific forward assay and single "
+            "reads, the read has to be mapped to the same strand as the feature. "
+            "For paired-end reads, the first read has to be on the same strand and "
+            "the second read on the opposite strand. In strand-specific reverse "
+            "assay these rules are reversed.",
+            default="non_specific",
+        )
+        cdna_index = DataField(
+            data_type="index:salmon",
+            label="Indexed cDNA reference sequence",
+            required=False,
+            description="Transcriptome index file created using the Salmon indexing tool. "
+            "cDNA (transcriptome) sequences used for index file creation must be "
+            "derived from the same species as the input sequencing reads to "
+            "obtain the reliable analysis results.",
+            hidden="assay_type != 'auto'",
+        )
+        rrna_reference = DataField(
+            data_type="index:star",
+            label="Indexed rRNA reference sequence",
+            description="Reference sequence index prepared by STAR aligner indexing tool.",
+        )
+        globin_reference = DataField(
+            data_type="index:star",
+            label="Indexed Globin reference sequence",
+            description="Reference sequence index prepared by STAR aligner indexing tool.",
+        )
+
+        class Preprocessing:
+            """Preprocessing with BBDuk."""
+
+            adapters = ListField(
+                inner=DataField(data_type="seq:nucleotide"),
+                label="Adapters",
+                required=False,
+                description="FASTA file(s) with adapters.",
+            )
+            custom_adapter_sequences = ListField(
+                inner=StringField(),
+                label="Custom adapter sequences",
+                required=False,
+                default=[],
+                description="Custom adapter sequences can be specified by inputting them "
+                "one by one and pressing Enter after each sequence.",
+            )
+            kmer_length = IntegerField(
+                label="K-mer length [k=]",
+                default=23,
+                description="Kmer length used for finding contaminants. "
```

Comment:
Sometimes we use `kmer` and other times `k-mer`. Make it uniform throughout the input fields.